clang/TranslationUnit.d

```diff
@@ -37,6 +37,61 @@ struct TranslationUnit
                 options));
     }
 
+    static TranslationUnit parseString (
+        Index index,
+        string source,
+        string[] commandLineArgs,
+        CXUnsavedFile[] unsavedFiles = null,
+        uint options = CXTranslationUnit_Flags.CXTranslationUnit_DetailedPreprocessingRecord)
+    {
+        import std.random;
+        import std.file;
+        import std.path;
+        import std.stdio;
+
+        string randstr (uint n)
+        {
+            string result;
+
+            for (uint i = 0; i < n; ++i)
+                result ~= uniform!("[]")('0', '9');
+
+            return result;
+        }
+
+        string tempdir ()
+        {
+            return tempDir() ~ dirSeparator ~ "dstep";
+        }
+
+        string tempname ()
+        {
+            return tempdir() ~ dirSeparator ~ randstr(20) ~ ".dstep.h";
+        }
+
+        string temp = tempdir();
+
+        if (!exists(temp)) 
+            mkdir(temp);
```

Comment:
If you consider current solution too risky, I can add some random suffix to `dstep` sub-directory so that there is no chance of collision with another application.